### PR TITLE
linkify: Add web+ schema support

### PR DESCRIPTION
**Pull request by brunnre8**
Replaces https://github.com/thelounge/thelounge/pull/4818

@SoniEx2 ping

### CODE REVIEW

File: shared/linkify.ts [hunk 1]
```diff
@@ -55,6 +55,27 @@ for (const schema of commonSchemes) {
 	linkify.add(schema + ":", "http:");
 }
 
+linkify.add("web+", {
+	validate(text: string, pos: number, self: LinkifyIt.LinkifyIt) {
+		const webSchemaRe = /^[a-z]+:/g;
```
Comment (SoniEx2) [inline shared/linkify.ts:60]: this should definitely be `i` - it's normalized by the browser. other than that the use of `g` and `lastIndex` looks fine...? not entirely sure, never used that before. (well, if the tests pass it's good right? .-.)

**Review by SoniEx2 — CHANGES_REQUESTED** [review]

File: shared/linkify.ts (hunk 1 above)
Comment (brunnre8) [reply shared/linkify.ts:60]: I don't want to match case insensitive no. It should accept valid web+ schemes, not some other things.

g is needed that last index gets set

File: shared/linkify.ts [hunk 2]
```diff
@@ -55,6 +55,27 @@ for (const schema of commonSchemes) {
...
+
+		const linkEnd = self.testSchemaAt(text, "http:", pos + webSchemaRe.lastIndex);
+
+		if (linkEnd === 0) {
+			return 0;
+		}
+
+		return webSchemaRe.lastIndex + linkEnd;
+	},
+	normalize(match) {
```
[the remark this answers is not in the document]
Comment (brunnre8) [reply shared/linkify.ts:74]: bloody monkey patching...
It's bad mkey (in my best Mr. Mackey voice)

[the remark this answers is not in the document]
Comment (brunnre8) [reply shared/linkify.ts:74]: better now?

[the remark this answers is not in the document]
Comment (SoniEx2) [reply shared/linkify.ts:74]: wow, that compiles?! how?

File: shared/linkify.ts (hunk 1 above)
Comment (SoniEx2) [reply shared/linkify.ts:60]: it'll also accept `WEB+ap://example.org`. you can add a test for that if you don't believe it.

this should definitely have an `i`.

**Review by SoniEx2 — CHANGES_REQUESTED** [review]

**Review by SoniEx2 — APPROVED** [review]
LGTU :+1:

File: shared/linkify.ts (hunk 1 above)
Comment (brunnre8) [reply shared/linkify.ts]: ack

File: shared/linkify.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (brunnre8) [reply shared/linkify.ts:74]: :shrug:  No idea. That whole code should be fixed up

**Pull request merged**